ip-handling/draft-ietf-rtcweb-ip-handling.xml

```diff
@@ -209,22 +212,33 @@ ipr="trust200902">
           path, and is ideal when media performance is the highest priority, but
           it discloses the most information.</t>
 
-          <t>Default route + associated local addresses: WebRTC MUST follow the
-          kernel routing table rules, which will typically cause media packets to
-          take the same route as the application's HTTP traffic. If an
-          enterprise TURN server is present, the preferred route MUST
-          be through this TURN server. Once an interface has been chosen,
-          the private IPv4 and IPv6 addresses associated with this
-          interface MUST be discovered and provided to the
-          application. This ensures that direct connections can still be
-          established in this mode.</t>
+          <t>Default route + privacy-preserving local addresses: WebRTC MUST
+          follow the kernel routing table rules, which will typically cause
+          media packets to take the same route as the application's HTTP
+          traffic. If an enterprise TURN server is present, the preferred
+          route MUST be through this TURN server. Once an interface has been
+          chosen:
+            <list style="symbols">
+              <t>If the interface has an associated <xref target="RFC4941" />
+              IPv6 address, it MUST be provided to the application.</t>
+              <t>If the interface has an associated <xref target="RFC1918" />
+              IPv4 address, and multicast DNS <xref target="RFC6762" /> is
+              supported, the private IPv4 address MUST be replaced with a
+              short-lived mDNS hostname. Any "srflx" candidates generated from
+              this interface MUST have their raddr fields set to 0.0.0.0.</t>
+            </list>
```

Comment:
what happens if the interface has only a public IPv4 address? The procedure doesn't say that you expose it, which means that no address will be exposed. This seems wrong.
(this was actually a problem with the previous text too)

Reply:
I think we have two options: 
1) if it is not 1918 (or 6598), expose it and don't bother with mDNS.
2) if we determine it is public (by virtue of getting a STUN server reply with this address), expose it.

1) seems considerably easier to implement.

Reply:
Youenn suggested today that we might want to do 2) for both IPv4 and IPv6 (to handle NAT64 cases). This is an interesting idea.